### PR TITLE
initial work on handling alice#bob implicit teams

### PR DESCRIPTION
**Pull request by maxtaco**
- change the AVDL
- change parsing
- add some tests
- there might be other features/aspects missing, but this will unblock the other PR I'm working on, which needs the vendor in the
protocol changes.

### CODE REVIEW

File: go/libkb/assertion.go [hunk 1]
```diff
@@ -620,30 +621,63 @@ func parseImplicitTeamPart(ctx AssertionContext, s string) (typ string, name str
...
+			continue
+		}
+		seen[idx] = true
+		if typ == "keybase" {
+			ret.KeybaseUsers = append(ret.KeybaseUsers, name)
+		} else {
+			ret.UnresolvedUsers = append(ret.UnresolvedUsers, sa)
+		}
 	}
+	sort.Slice(ret.KeybaseUsers, func(i, j int) bool { return ret.KeybaseUsers[i] < ret.KeybaseUsers[j] })
```
Comment (patrickxb) [inline go/libkb/assertion.go]: `sort.Strings(ret.KeybaseUsers)`

**Comment by keybase-ci-visdiff** [comment]
The commits a379410c7ed5...02c829c4a0d6 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503331891778-a379410c7ed5-02c829c4a0d6-linux/Markdown-emoji-a379410c7ed5.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503331891778-a379410c7ed5-02c829c4a0d6-linux/Markdown-emoji-02c829c4a0d6.png)  
   Markdown-emoji comparison
</summary>

File: go/libkb/assertion.go [hunk 2]
```diff
@@ -676,7 +676,7 @@ func parseImplicitTeamUserSet(ctx AssertionContext, s string, seen map[string]bo
 			ret.UnresolvedUsers = append(ret.UnresolvedUsers, sa)
 		}
 	}
-	sort.Slice(ret.KeybaseUsers, func(i, j int) bool { return ret.KeybaseUsers[i] < ret.KeybaseUsers[j] })
+	sort.Slice(ret.KeybaseUsers)
```
Comment (patrickxb) [inline go/libkb/assertion.go]: sort.Strings(ret.KeybaseUsers)

(Strings, not Slice)

**Comment by maxtaco** [comment]
thx, fixed the second time around

**Comment by keybase-ci-visdiff** [comment]
The commits a379410c7ed5...d22dab135ad1 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503333300262-a379410c7ed5-d22dab135ad1-linux/Markdown-emoji-a379410c7ed5.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503333300262-a379410c7ed5-d22dab135ad1-linux/Markdown-emoji-d22dab135ad1.png)  
   Markdown-emoji comparison
</summary>

**Pull request merged**